tutorials/environmental_data_and_visualization.md

```diff
@@ -0,0 +1,173 @@
+sample points, Gazebo will interpolate the data.
+
+## Data Format
+
+As mentioned earlier, the data should be a CSV file with grid based spacing.
+The first line of the CSV file should be the heading for each column.
+There should be a column for time (in seconds), the x coordinate, y coordinate
+and z coordinate (in meters). You may have multiple columns of data like so:
+```
+timestamp,humidity,x,y,z
+0,0,-1,-1,-1
+0,0,-1,-1, 1
+0,0,-1, 1,-1
+0,0,-1, 1, 1
+0,0, 1,-1,-1
+0,0, 1,-1, 1
+0,0, 1, 1,-1
+0,0, 1, 1, 1
+1,1,-1,-1,-1
+1,1,-1,-1, 1
+1,1,-1, 1,-1
+1,1,-1, 1, 1
+1,1, 1,-1,-1
+1,1, 1,-1, 1
+1,1, 1, 1,-1
+1,1, 1, 1, 1
+
+```
+For the x and y coordinates you may also use latitude and longitude in degrees.
+Heading names don't matter. Order also doesn't matter.
+```
+timestamp,humidity,latitude,longitude,altitude
+1658923062,91,36.80029505,-121.788972517,0.8
+1658923062,88,36.80129505,-121.788972517,0.8
+1658923062,89,36.80029505,-121.789972517,0.8
+1658923062,92,36.80129505,-121.789972517,0.8
+1658923063,90,36.80029505,-121.788972517,0.8
+1658923063,85,36.80129505,-121.788972517,0.8
+1658923063,87,36.80029505,-121.789972517,0.8
+1658923063,94,36.80129505,-121.789972517,0.8
+```
+> :warning: Note that only one Environment Data file per world is allowed. Also,
+Gazebo can only load data that fits into memory.
+
+## Pre-Loading
+If you are writing SDFormat files by hand or via a script you may want to
+pre-load the data. You may use the `EnvironmentPreload` plugin for this. For
+instance we could load the first file like so (by adding the environment to the
+):
+```xml
+    <plugin
+      filename="gz-sim-environment-preload-system"
+      name="gz::sim::systems::EnvironmentPreload">
+      <data>environmental_data.csv</data>
+      <dimensions>
+        <time>timestamp</time>
+        <space>
+          <x>x</x>
+          <y>y</y>
+          <z>z</z>
+        </space>
+      </dimensions>
+    </plugin>
+```
+If you would like to use Latitude and Longitude you may set the `reference` to
+`spherical` and the `units` to `degrees`.
+```xml
+    <plugin
+      filename="gz-sim-environment-preload-system"
+      name="gz::sim::systems::EnvironmentPreload">
+      <data>environmental_data.csv</data>
+      <dimensions>
+        <time>timestamp</time>
+        <space reference="spherical" units="degrees">
+          <x>latitude</x>
+          <y>longitude</y>
+          <z>altitude</z>
+        </space>
+      </dimensions>
+    </plugin>
+```
+
+## Via GUI
+To load the data, click on the 3 dots and search for the "Environment Loader":
+@image html files/environment_data/environment_loader.png
+
+Click on it and you will see the environment loader:
+
+@image html files/environment_data/environment_loader_options.png
+
+Select the file you would want to load by clicking on the arrow next to the
+field named "Data file path".
+
+@image html files/environment_data/environment_file_loaded.png
+
+Select the appropriate fields and then click "load".
```

Comment:
Currently, the data is not loading for me. When I select the `environmental_data.csv` in `src/examples/worlds` and click Load, I get this error
```
[GUI] [Err] [EnvironmentLoader.cc:194] No environmental data file was found at 
```

Reply:
Thanks for checking this. Were you using these two branches:
* https://github.com/gazebosim/gz-sim/pull/1842
* https://github.com/gazebosim/gz-msgs/pull/320

Reply:
Yes, those were the branches I used. I also tried after merging from `gz-msgs9` and `gz-sim7`, as the gz-msgs PR says the branch is outdated.

Reply:
Apologies for not getting back to this quicker but I've found the reason why. I've fixed it with c30639ab4.